shared/js/https.js

```diff
@@ -1,228 +1,211 @@
 (function () {
-const db = require('db')
+const load = require('load')
 const settings = require('settings')
 const utils = require('utils')
-let knownMixedContentList
 
-settings.ready().then(() => {
-    load.JSONfromLocalFile(constants.httpsWhitelist, (wl) => knownMixedContentList = wl)
-})
+// number of chunks to split up the data
+// when storing to Safari to avoid over quota issues:
+const LOCAL_STORAGE_CHUNKS = 5
+
+let httpsUpgradeList = []
 
 class HTTPS {
 
     constructor () {
-        this.db = null
-        this.dbObjectStore = 'https'
-
-        // if this.isSync: load Chrome-only synchronous rules object into memory
-        this.isSync = utils.isChromeBrowser() ? true : false
-        this.syncRuleCache = {}
-
         this.isReady = false
-        this._ready = this.init().then(() => this.isReady = true)
+        this.init().then(() => this.isReady = true)
+
+        // in the background, try to update the list:
+        this.updateList()
 
         return this
     }
 
     init () {
-        return new Promise((resolve) => {
-            db.ready().then(() => {
-                this.db = db
+        console.log("HTTPS: init()")
 
-                if (this.isSync) {
-                    this.initSync().then(() => resolve())
-                } else {
-                    resolve()
-                }
+        return new Promise((resolve) => {
+            // Try to load the list from local storage
+            this.getFromStorage().then((items) => {
+
+                // if items were found in local storage,
+                console.log("HTTPS: init() found existing list in storage with " + items.length + " items")
+                httpsUpgradeList = items
+                return resolve()
+
+            }, () => {
+
+                // if it fails to find anything in local storage,
```

Comment:
Again, are we sure we want to do this? We get no recourse if something is broken in there. We literally have to wait for a release. The whitelist was different because it was the other way, explicitly saying ignore this one.

Reply:
agree, will remove.